src/Strategy/Perl.hs

```diff
@@ -0,0 +1,202 @@
+
+-- | Represents Metafile for various versions.
+-- References:
+-- v1.0: http://module-build.sourceforge.net/META-spec-v1.0.html
+-- v1.1: http://module-build.sourceforge.net/META-spec-v1.1.html
+-- v1.2: http://module-build.sourceforge.net/META-spec-v1.2.html
+-- v1.3: http://module-build.sourceforge.net/META-spec-v1.3.html
+-- v1.4: https://metacpan.org/release/DAGOLDEN/CPAN-Meta-2.101090
+-- v2.0: https://metacpan.org/release/DAGOLDEN/CPAN-Meta-2.150010
+data PerlMeta = PerlMeta
+  { version :: Double
+  , runtimeRequires :: Maybe (Map PackageName (Maybe Text))
+  , buildRequires :: Maybe (Map PackageName (Maybe Text))
+  , testRequires :: Maybe (Map PackageName (Maybe Text))
+  , developRequires :: Maybe (Map PackageName (Maybe Text))
+  , configureRequires :: Maybe (Map PackageName (Maybe Text))
+  }
+  deriving (Generic, Show, Eq, Ord)
+
+instance FromJSON PackageName where
+  parseJSON (AesonTypes.String packageName) = pure $ PackageName packageName
+  parseJSON _ = fail "failed to parse package's name"
+
+instance FromJSON PerlMeta where
+  parseJSON = withObject "meta content" $ \o -> do
+    -- spec_version can be either be string or number
+    -- in yaml, version is provided as string, where as in json, it is numeric
+    specVersion :: Double <-
+      (o .: "meta-spec" |> "version")
+        <|> ( do
+                v <- o .: "meta-spec" |> "version"
+                case readMaybe v of
+                  Nothing -> fail ("Expected numeric value for version field, but got: " <> show v)
+                  Just x -> pure x
+            )
+
+    if specVersion > 1.4
+      then parseAboveV1_4 o specVersion
+      else parseBelowV1_5 o specVersion
+    where
+      (|>) :: FromJSON a => Parser Object -> Text -> Parser a
+      (|>) parser key = do
+        obj <- parser
+        obj .: key
+
+      (|?>) :: FromJSON a => Parser (Maybe Object) -> Text -> Parser (Maybe a)
+      (|?>) parser key = do
+        obj <- parser
+        case obj of
+          Nothing -> pure Nothing
+          Just o -> o .:? key
+
+      -- Reference: https://metacpan.org/release/DAGOLDEN/CPAN-Meta-2.101090
+      parseAboveV1_4 obj version = do
+        runtimeRequires <- obj .:? "prereqs" |?> "runtime" |?> "requires"
+        buildRequires <- obj .:? "prereqs" |?> "build" |?> "requires"
+        testRequires <- obj .:? "prereqs" |?> "test" |?> "requires"
+        developRequires <- obj .:? "prereqs" |?> "develop" |?> "requires"
+        configureRequires <- obj .:? "prereqs" |?> "configure" |?> "requires"
+
+        pure $
+          PerlMeta
+            version
+            runtimeRequires
+            buildRequires
+            testRequires
+            developRequires
+            configureRequires
+
+      -- Reference: http://module-build.sourceforge.net/META-spec-v1.4.html
+      parseBelowV1_5 obj version = do
+        runtimeRequires1 <- obj .:? "requires"
```

Comment:
I don't think you need to make this `runtimeRequires1` since it's in a different do block from the one above.
```suggestion
        runtimeRequires <- obj .:? "requires"
```